sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1021,6 +1021,19 @@ class Analyzer(
         case e: Expand =>
           failOnOuterReferenceInSubTree(e, "an EXPAND")
           e
+        case l @ LocalLimit(_, _) =>
```

Comment:
Style: use `l: LocalLimit` instead of `l @ LocalLimit(_, _)` it makes it a bit more readable. Same for `GlobalLimit` and `Sample`.